From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Benedikt Niedermayr <benedikt.niedermayr@siemens.com>
Date: Fri, 12 Apr 2024 10:22:49 +0200
Subject: [PATCH] gpio-chardev: init compatibility

Pins initialized with mraa_gpio_chardev_init() should behave more
like pins initialized with mraa_gpio_init(). This applies
especially for several attributes of the
pin's mraa_gpio_context struct:

* advance_func: Required to execute custom functions of the pin's
                function table.
* phy_pin: Required to distinguish between board pins and onboard
           mulitplexer pins(pull_en, out_en)

Signed-off-by: Benedikt Niedermayr <benedikt.niedermayr@siemens.com>
---
 src/gpio/gpio.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/gpio/gpio.c b/src/gpio/gpio.c
index f13081f514af..a0c56c1e7e16 100644
--- a/src/gpio/gpio.c
+++ b/src/gpio/gpio.c
@@ -505,6 +505,16 @@ mraa_gpio_chardev_init(int pins[], int num_pins)
     /* Initialize events array. */
     dev->events = NULL;
 
+    /*
+     * Single pin init
+     * Stay compatible to mraa_gpio_init()
+     */
+    if (num_pins == 1) {
+        dev->advance_func = board->adv_func;
+        dev->pin = pins[0];
+        dev->phy_pin = dev->pin;
+    }
+
     return dev;
 }
 
